usaspending_api/awards/v2/views/award_spending.py

```diff
@@ -74,17 +74,19 @@ def get_queryset(self):
         top_tier_agency_id = Agency.objects.filter(id=awarding_agency_id).first().toptier_agency_id
         queryset = Transaction.objects.all()
 
-        # Filter based on fiscal_year, awarding_agency_id, and category
         queryset = queryset.filter(
             fiscal_year=fiscal_year,
-            awarding_agency__toptier_agency=top_tier_agency_id,
-            award__category=award_category
+            awarding_agency__toptier_agency=top_tier_agency_id
         ).annotate(
             award_category=F('award__category'),
             recipient_id=F('recipient__legal_entity_id'),
             recipient_name=F('recipient__recipient_name')
         )
 
+        if award_category is not None:
+            # Filter based on fiscal_year, awarding_agency_id, award_category
```

Comment:
We should update this comment, and also move parts of it to the right place (after line 77)